railgun/manager/prometheus.go

```diff
@@ -0,0 +1,102 @@
+package manager
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kong/kubernetes-ingress-controller/railgun/internal/ctrlutils"
+	"github.com/kong/kubernetes-ingress-controller/railgun/internal/mgrutils"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+func RunHTTP(log logr.Logger) {
+	stopCh := make(chan struct{})
+	var wg sync.WaitGroup
+	mux := http.NewServeMux()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		ServeHTTP(true,
+			ctrlutils.PROMTHPORT, mux, stopCh,
+			log,
+			&wg)
+	}()
+}
+
+// ServeHTTP enable HTTP Server
+// - prometheus
+func ServeHTTP(enableProfiling bool,
+	port int,
+	mux *http.ServeMux,
+	stop <-chan struct{},
+	logger logr.Logger,
+	wg *sync.WaitGroup) {
+	defer wg.Done()
+	mux.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Railgun has a healthz check already - here is a test that verifies that:

https://github.com/Kong/kubernetes-ingress-controller/blob/bc19341034eea5c639a49155173704ef87a4d4b7/test/integration/controller_test.go#L17

There's no need to add another one.